srcs/builtins/cd_builtin.c

```diff
@@ -1,67 +1,51 @@
 /* ************************************************************************** */
 /*                                                                            */
 /*                                                        :::      ::::::::   */
-/*   cd.c                                               :+:      :+:    :+:   */
+/*   cd_builtin.c                                       :+:      :+:    :+:   */
 /*                                                    +:+ +:+         +:+     */
-/*   By: dkolida <dkolida@student.42warsaw.pl>      +#+  +:+       +#+        */
+/*   By: dmodrzej <dmodrzej@student.42.fr>          +#+  +:+       +#+        */
 /*                                                +#+#+#+#+#+   +#+           */
 /*   Created: 2024/07/20 23:02:48 by dmodrzej          #+#    #+#             */
-/*   Updated: 2024/07/21 19:56:11 by dkolida          ###   ########.fr       */
+/*   Updated: 2024/07/24 23:20:23 by dmodrzej         ###   ########.fr       */
 /*                                                                            */
 /* ************************************************************************** */
 
 #include "minishell.h"
 
-char	*get_env_var(char **env, char *var)
+static int	path_error(char *path)
 {
-	int	i;
-	int	len;
+	ft_putstr_fd("cd: ", 2);
+	perror(path);
+	free(path);
+	return (1);
+}
 
-	i = 0;
-	len = ft_strlen(var);
-	while (env[i])
-	{
-		if (!ft_strncmp(env[i], var, len) && env[i][len] == '=')
-			return (env[i] + len + 1);
-		i++;
-	}
-	return (NULL);
+static int	home_error(void)
+{
+	ft_putstr_fd("cd: HOME not set\n", 2);
+	return (1);
 }
 
-// #TODO Need to be refactored according to Norminette
-/*
-int	builtin_cd(t_shell *shell, char **args)
+int	cd_builtin(t_shell *shell, char **args)
 {
 	char	*path;
 
 	if (!args[1])
 	{
-		path = ft_strdup(get_env_var(shell->env, "HOME"));
+		path = ft_strdup(get_env_var(shell, "HOME"));
 		if (!path)
-		{
-			ft_putstr_fd("cd: HOME not set\n", 2);
-			return (1);
-		}
+			return (home_error());
 	}
 	else if (args[1][0] == '~')
 	{
-		path = ft_strjoin(get_env_var(shell->env, "HOME"), args[1] + 1);
+		path = ft_strjoin(get_env_var(shell, "HOME"), args[1] + 1);
 		if (!path)
-		{
-			ft_putstr_fd("cd: HOME not set\n", 2);
-			return (1);
-		}
+			return (home_error());
 	}
 	else
 		path = ft_strdup(args[1]);
```

Comment:
That if path will be equal to NULL? on path = ft_strdup(args[1]); ?

Reply:
ft_strdup is protected against that:
	if (!s)
		return (0);
so after that in cd_builtin:
	if (!path)
		return (1);
basically, no duplication and ending cd function with status 1